Sources/SWBWebAssemblyPlatform/Plugin.swift

```diff
@@ -0,0 +1,144 @@
+        ["WEBASSEMBLY_DEPLOYMENT_TARGET"]
+    }
+
+    func additionalPlatforms() -> [(path: Path, data: [String: PropertyListItem])] {
+        [
+            (.root, [
+                "Type": .plString("Platform"),
+                "Name": .plString("webassembly"),
+                "Identifier": .plString("webassembly"),
+                "Description": .plString("webassembly"),
+                "FamilyName": .plString("WebAssembly"),
+                "FamilyIdentifier": .plString("webassembly"),
+                "IsDeploymentPlatform": .plString("YES"),
+            ])
+        ]
+    }
+}
+
+struct WebAssemblySDKRegistryExtension: SDKRegistryExtension {
+    func additionalSDKs(platformRegistry: PlatformRegistry) async -> [(path: Path, platform: SWBCore.Platform?, data: [String: PropertyListItem])] {
+        guard let host = try? ProcessInfo.processInfo.hostOperatingSystem() else {
+            return []
+        }
+
+        guard let wasmPlatform = platformRegistry.lookup(name: "webassembly") else {
+            return []
+        }
+
+        let defaultProperties: [String: PropertyListItem] = [
+            "SDK_STAT_CACHE_ENABLE": "NO",
+
+            // Workaround to avoid `-add_ast_path` on WebAssembly, apparently this needs to perform some "swift modulewrap" step instead.
+            "GCC_GENERATE_DEBUGGING_SYMBOLS": .plString("NO"),
+
+            // Workaround to avoid `-dependency_info` on WebAssembly.
+            "LD_DEPENDENCY_INFO_FILE": .plString(""),
+
+            // WebAssembly uses wasm-ld, not the Apple linker
+            "LD_DETERMINISTIC_MODE": "NO",
+            // HACK: The following setting does not work. Need to set `ENABLE_TESTABILITY` to `NO`
+            // // Workaround to avoid `-export-dynamic` on WebAssembly.
+            // "LD_EXPORT_GLOBAL_SYMBOLS": "NO",
+
+            "GENERATE_TEXT_BASED_STUBS": "NO",
+            "GENERATE_INTERMEDIATE_TEXT_BASED_STUBS": "NO",
+
+            "CHOWN": "/usr/bin/chown",
+
+            "LIBTOOL": .plString(host.imageFormat.executableName(basename: "llvm-lib")),
+            "AR": .plString(host.imageFormat.executableName(basename: "llvm-ar")),
+        ]
+
+        // Map triple to parsed triple components
+        let supportedTriples: [String: (arch: String, os: String, env: String?)] = [
+            "wasm32-unknown-wasi": ("wasm32", "wasi", nil),
+            "wasm32-unknown-wasip1": ("wasm32", "wasip1", nil),
+            "wasm32-unknown-wasip1-threads": ("wasm32", "wasip1", "threads"),
+        ]
+
+        let wasmSwiftSDKs = (try? SwiftSDK.findSDKs(
+            targetTriples: Array(supportedTriples.keys),
+            fs: localFS
+        )) ?? []
+
+        var wasmSDKs: [(path: Path, platform: SWBCore.Platform?, data: [String: PropertyListItem])] = []
+
+        for wasmSDK in wasmSwiftSDKs {
+            for (triple, tripleProperties) in wasmSDK.targetTriples {
+                guard let (arch, os, env) = supportedTriples[triple] else {
```

Comment:
Interesting approach, defining a dedicated SDK for each triple. I wonder if we should handle this somewhat differently, by adding all the triples to a single SDK and then having dedicated build settings that control which ABI variant you're building against?

@owenv, any thoughts here?

Reply:
It's an interesting question, since I think right now WebAssembly doesn't have anything like a 'universal' binary. I'm not sure I have a good answer right now

Reply:
> by adding all the triples to a single SDK and then having dedicated build settings that control which ABI variant you're building against?

I agree that it's more straightforward mapping from Swift SDK to SwiftBuild SDK.

However, what is holding me back from adopting this approach is that I don't know what to assign to the variant's name. In the Swift SDK, target triple is the key for each variant,  but I am wondering if using target triple as SDK_VARIANT is the natural way for SwiftBuild as other variants are `macosx`, `iosmac`, and so on.

Reply:
SDK_VARIANT is not very flexible today, but it we're able to lift some of its limitations it could be a good fit

Reply:
Seems fine for now, we can always revisit as we get a better feel for how this and Swift SDKs fit together with Swift Build